src/test/scala/chiselTests/ConnectSpec.scala

```diff
@@ -193,4 +201,35 @@ class ConnectSpec extends ChiselPropSpec with Utils {
     val expectedTypeMismatchError = """.*@: Sink \(UInt<4>\) and Source \(SInt<4>\) have different types."""
     (typeMismatchError.getMessage should fullyMatch).regex(expectedTypeMismatchError)
   }
+
+  // (S)trict Connect tests
+  property("(S.a) SInt :<>= SInt should succeed") {
+    ChiselStage.elaborate { new CrossStrictConnects(SInt(16.W), SInt(16.W)) }
+  }
+  property("(S.b) UInt :<>= UInt should succeed") {
+    ChiselStage.elaborate { new CrossStrictConnects(UInt(16.W), UInt(16.W)) }
+  }
+  property("(S.c) SInt :<>= UInt should fail") {
+    intercept[ChiselException] { ChiselStage.elaborate { new CrossStrictConnects(UInt(16.W), SInt(16.W)) } }
+  }
+  property("(S.d) Decoupled :<>= Decoupled should succeed") {
+    class Decoupled extends Bundle {
+      val bits = UInt(3.W)
+      val valid = Bool()
+      val ready = Flipped(Bool())
+    }
+    val out = ChiselStage.emitChirrtl { new CrossStrictConnects(new Decoupled, new Decoupled) }
+    assert(out.contains("io.out <= io.in"))
+  }
```

Comment:
I'd like to see 2 other tests
1. Non-commutativity of `:<>=`, eg. `io.in :<>= io.out` in `CrossStrictConnects` would fail
2. Connecting 2 wires with `:<>=`

Reply:
aren't all `Wire`s able to exhibit FIRRTL's duplex flow?

Reply:
I added a bunch more tests including these two cases as well as Vectors of same and different lengths